yt_dlp/extractor/rumble.py

```diff
@@ -62,26 +123,48 @@ def _extract_embed_urls(cls, url, webpage):
     def _real_extract(self, url):
         video_id = self._match_id(url)
         video = self._download_json(
-            'https://rumble.com/embedJS/', video_id,
-            query={'request': 'video', 'v': video_id})
-        title = unescapeHTML(video['title'])
+            'https://rumble.com/embedJS/u3/', video_id,
+            query={'request': 'video', 'ver': 2, 'v': video_id})
+
+        sys_msg = traverse_obj(video, ('sys', 'msg'), default=None)
+        if sys_msg:
+            self.report_warning(sys_msg, video_id=video_id)
+
+        if video.get('live') == 0:
+            live_status = 'not_live' if video.get('livestream_has_dvr') is None else 'was_live'
+        elif video.get('live') == 1:
+            live_status = 'is_upcoming' if video.get('livestream_has_dvr') else 'was_live'
+        elif video.get('live') == 2:
+            live_status = 'is_live'
+        else:
+            live_status = None
 
         formats = []
-        for height, ua in (video.get('ua') or {}).items():
-            for i in range(2):
-                f_url = try_get(ua, lambda x: x[i], compat_str)
-                if f_url:
-                    ext = determine_ext(f_url)
-                    f = {
-                        'ext': ext,
-                        'format_id': '%s-%sp' % (ext, height),
-                        'height': int_or_none(height),
-                        'url': f_url,
-                    }
-                    bitrate = try_get(ua, lambda x: x[i + 2]['bitrate'])
-                    if bitrate:
-                        f['tbr'] = int_or_none(bitrate)
-                    formats.append(f)
+        for ext, ext_info in (video.get('ua') or {}).items():
+            if not ext_info:
+                continue
+            for height, video_info in (ext_info or {}).items():
+                meta = video_info.get('meta') or {}
+                if 'url' not in video_info:
+                    continue
+                if ext == 'hls':
+                    formats.extend(
+                        self._extract_m3u8_formats(
+                            video_info['url'], video_id, ext='mp4', m3u8_id='hls', fatal=False))
+                    if meta.get('live') is True and video.get('live') == 1:
+                        live_status = 'post_live'
+                    continue
+                fmt = {
+                    'ext': ext,
+                    'url': video_info['url'],
+                    'format_id': '%s-%sp' % (ext, height),
+                    'height': int_or_none(height),
+                    'fps': video.get('fps'),
+                }
+                fmt.update(
+                    {key: meta[meta_key] for meta_key, key in self.FORMAT_MAPPING.items()
+                     if meta_key in meta})
+                formats.append(fmt)
```

Comment:
```suggestion
                formats.append({
                    'ext': ext,
                    'url': video_info['url'],
                    'format_id': '%s-%sp' % (ext, height),
                    'height': int_or_none(height),
                    'fps': video.get('fps'),
                    **traverse_obj(meta, {
                        'tbr': 'bitrate',
                        'filesize': 'size',
                        'width': 'w',
                        'height': 'h',
                    }, default={})
                })
```
or since the keys are simple enough, just:
```suggestion
                formats.append({
                    'ext': ext,
                    'url': video_info['url'],
                    'format_id': '%s-%sp' % (ext, height),
                    'height': int_or_none(meta.get('h') or height),
                    'fps': video.get('fps'),
                    'tbr': meta.get('bitrate'),
                    'filesize': meta.get('size'),
                    'width': meta.get('w'),
                })
```